controllers/servers/csi/csi_controller_server.py

```diff
@@ -75,19 +75,28 @@ def CreateVolume(self, request, context):
                     logger.debug("requested size is 0 so the default size will be used : {0} ".format(
                         required_bytes))
 
-                if volume_parameters.virt_snap_func and source_id:
-                    array_mediator.validate_space_efficiency_matches_source(space_efficiency, source_id, source_type)
+                is_virt_snap_func = volume_parameters.virt_snap_func
+                if is_virt_snap_func and source_id:
+                    source_object = array_mediator.get_object_by_id(source_id, source_type, is_virt_snap_func)
+                    if not source_object:
+                        handle_exception("source {}: {} not found".format(source_type, source_id), context,
+                                         grpc.StatusCode.NOT_FOUND, csi_pb2.CreateVolumeResponse)
+                    if source_type == config.SNAPSHOT_TYPE_NAME:
+                        source_object = array_mediator.get_object_by_id(source_object.source_id,
+                                                                        config.VOLUME_TYPE_NAME)
```

Comment:
```suggestion
                        snapshot_source = array_mediator.get_object_by_id(source_object.source_id,
                                                                          config.VOLUME_TYPE_NAME)
```

Reply:
it can be a volume source..

Reply:
done

Reply:
```suggestion
                        source_volume = array_mediator.get_object_by_id(source_object.source_id,
                                                                        config.VOLUME_TYPE_NAME)
                    else:
                        source_volume = source_object
```
would this be more readable than current diff?

Reply:
I dont understand your suggestion

Reply:
I'm looking for a way to make this:
`utils.validate_parameters_match_source_volume(space_efficiency, required_bytes, source_object)`
into this:
`utils.validate_parameters_match_source_volume(space_efficiency, required_bytes, source_volume)`